### PR TITLE
Add and Fetch credential transaction rewrite

### PR DESCRIPTION
**Pull request by akash1729**

### CODE REVIEW

File: dtos/user_dto.go [hunk 1]
```diff
@@ -13,6 +11,6 @@ type Login struct {
 }
 
 type LoginReturn struct {
```
Comment (AbrahamGeorge8547) [inline dtos/user_dto.go:13]: type is user for login return and not string

File: controllers/credential_controller.go [hunk 2]
```diff
@@ -17,15 +19,34 @@ func AddCredential(ctx *gin.Context) {
 		return
 	}
 
-	userIdInterface, _ := ctx.Get("userId")
-	userID, _ := userIdInterface.(uuid.UUID)
-	credentialId, err := service.CreateCredential(ctx, req, userID)
+	createdUserIDInterface, _ := ctx.Get("userId")
```
Comment (AbrahamGeorge8547) [inline controllers/credential_controller.go:22]: why createdUserIDInterface

File: controllers/credential_controller.go [hunk 3]
```diff
@@ -17,15 +19,34 @@ func AddCredential(ctx *gin.Context) {
...
 	SendResponse(ctx, 201, credentialId, "Added Credential", nil)
 }
 
+func GetCredentialByID(ctx *gin.Context) {
+	userIdInterface, _ := ctx.Get("userId")
+	userID, _ := userIdInterface.(uuid.UUID)
+	credentialIDStr := ctx.Param("id")
+	credentailaID, _ := uuid.Parse(credentialIDStr)
+	credential, err := service.FetchCredentialByID(ctx, credentailaID, userID)
+	if err != nil {
```
Comment (AbrahamGeorge8547) [inline controllers/credential_controller.go:39]: is this user not authenticated?

File: dtos/user_dto.go (hunk 1 above)
Comment (akash1729) [reply dtos/user_dto.go:13]: yup. I will fix it

**Pull request merged**